mountpoint-s3/scripts/fs_bench.sh

```diff
@@ -45,43 +46,43 @@ run_fio_job() {
   job_name=$(basename "${job_file}")
   job_name="${job_name%.*}"
 
-  for i in $(seq 1 $iteration);
+  echo -n "Running job ${job_name} for ${iterations} iterations... "
+
+  for i in $(seq 1 $iterations);
   do
+    echo -n "${i};"
     fio --thread \
-      --output=${results_dir}/${job_name}_${i}.json \
+      --output=${results_dir}/${job_name}_iter${i}.json \
       --output-format=json \
       --directory=${mount_dir} \
       --filename=${bench_file} \
+      --eta=never \
       ${job_file}
   done
+  echo "done"
 
   # combine the results and find an average value
   jq -n 'reduce inputs.jobs[] as $job (null; .name = $job.jobname | .len += 1 | .value += (if ($job."job options".rw == "read")
       then $job.read.bw / 1024
       elif ($job."job options".rw == "randread") then $job.read.bw / 1024
       elif ($job."job options".rw == "randwrite") then $job.write.bw / 1024
-      else $job.write.bw / 1024 end)) | {name: .name, value: (.value / .len), unit: "MiB/s"}' ${results_dir}/${job_name}_*.json | tee ${results_dir}/${job_name}_parsed.json
-
-  # delete the raw output files
-  for i in $(seq 1 $iteration);
-  do
-    rm ${results_dir}/${job_name}_${i}.json
-  done
+      else $job.write.bw / 1024 end)) | {name: .name, value: (.value / .len), unit: "MiB/s"}' ${results_dir}/${job_name}_iter*.json | tee ${results_dir}/${job_name}_parsed.json
 }
 
-read_bechmark () {
+read_benchmark () {
   jobs_dir=mountpoint-s3/scripts/fio/read
 
   for job_file in "${jobs_dir}"/*.fio; do
     mount_dir=$(mktemp -d /tmp/fio-XXXXXXXXXXXX)
 
-    echo "Running ${job_name}"
-
     # mount file system
-    cargo run --release ${S3_BUCKET_NAME} ${mount_dir} \
+    set +e
```

Comment:
I am not very sure about the usage of set -e. Having set +e turns exit on non-zero return off right? Should not that be default?

Reply:
Exit on non-zero is not on by default (which always surprises me!).

With these changes, I enable exit on non-zero for the whole script. However for this section, I want to tolerate a non-zero exit code and capture it, so that I can provide a more informative error message. I allow non-zero exits using `set +e` and then turn it off again once the mount has succeeded using `set -e`.